To-Do List tasks application

Leon-Mbegera opened the pull request:
This is a useful tool in the form of an application to help its users personally arrange their daily/weekly tasks into lists/project.
- The todos can be assigned details e.g. title, description, due date, and priority status.
- the user is also given the ability to edit their todos, or even delete them.

Leon-Mbegera committed:
Add initial config files

Leon-Mbegera committed:
Add file to ignore node modules in the project

Leon-Mbegera committed:
Generate a package.json file with project specifications

Leon-Mbegera committed:
Add scripts necessary to build this project in package.json file

Leon-Mbegera committed:
Include webpack related packages in devDependencies

Leon-Mbegera committed:
Create src and dist folders, plus link the main.js file

Leon-Mbegera committed:
Create webpack.config.js file

Leon-Mbegera committed:
Write a simple alert statement and display on browser to make sure everything is working correctly

Leon-Mbegera committed:
Define method to create a new project

Leon-Mbegera committed:
Add functions to add and delete tasks from project

Leon-Mbegera committed:
Refactor functions into classes instead

Leon-Mbegera committed:
Initialize todo class with constructor with the todo properties

Leon-Mbegera committed:
Construct class to create a new todo object

Leon-Mbegera committed:
Define function to create a card for each project listed.

Leon-Mbegera committed:
Loop through each project from projects array and display it in projects section

Leon-Mbegera committed:
Include bootstrap link and stylesheet for styling

Leon-Mbegera committed:
Add an event listener for project creation and push it into all projects

Leon-Mbegera committed:
Create input fields to make a new todo

Leon-Mbegera committed:
Add event listener to submit new task onto a project

Leon-Mbegera committed:
Migrate class constructors into their own module

Leon-Mbegera committed:
Correct mode of importing normal exported constructors

Leon-Mbegera committed:
Create new todo creation form and append it to template

Leon-Mbegera committed:
Show a particular project todos

Leon-Mbegera committed:
Add button to modify a particular todo's details

Leon-Mbegera committed:
Style my todo list header

Leon-Mbegera committed:
Display default project and make others clickable

Leon-Mbegera committed:
Append a todo to the todo section with all its properties

Leon-Mbegera committed:
Modify the new todo creation form

Leon-Mbegera committed:
Create a new todo and add onto a particular project

Leon-Mbegera committed:
Display a clicked projects todo

Leon-Mbegera committed:
Show form to add new todo to a project, below its already existing todos

Leon-Mbegera committed:
Arrange a project's todos in a row

Leon-Mbegera committed:
Build form to modify a todo's details

Leon-Mbegera committed:
function to intake modified values for a todo task

Leon-Mbegera committed:
Input new values to modify a todo task

Leon-Mbegera committed:
Add button to delete a project todo

Leon-Mbegera committed:
Include a select to give todo a priority rank

Leon-Mbegera committed:
Put todo modify form in place of new todo creation form

Leon-Mbegera committed:
Style new todo input form

Leon-Mbegera committed:
Change todo modify form button to 'edit'

Leon-Mbegera committed:
Style the todo edit and delete buttons

Leon-Mbegera committed:
Block user from creating projects with duplicate names

Leon-Mbegera committed:
Display project todos in adjacent cards

Leon-Mbegera committed:
Add every new project to local storage API

Leon-Mbegera committed:
Replace a todo modify form with todo creation form after updating a task

Leon-Mbegera committed:
Get all projects stored in local storage

Leon-Mbegera committed:
Retrieve all projects from local storage before displaying them

Leon-Mbegera committed:
Retrieve all todos with their projects from local storage and displaying them

Leon-Mbegera committed:
Update local storage after modifying a todo

Leon-Mbegera committed:
Display a new set of todos after deletion

Leon-Mbegera committed:
Modify readme file with info and project instructions

Leon-Mbegera committed:
Onset of fixing eslint offenses

Leon-Mbegera committed:
Put project and class constructors into separate modules

Leon-Mbegera committed:
Define and invoke functions in correct order

Leon-Mbegera committed:
show a new todo creation form after every projects todos

Leon-Mbegera committed:
Resolve display todo creation form after displaying a task

Leon-Mbegera committed:
Fix eslint code offenses

Leon-Mbegera committed:
Commit code changes in todos branch before merge

Leon-Mbegera committed:
Merge code changes made in fixes branch into todos

afanimartin commented on src/index.js:
```diff
@@ -0,0 +1,255 @@
+const todoSection = document.getElementById('project-todos');
+const projectTodos = document.getElementById('todo-creation-form');
+
+// Loop thru each project and display it
```
Remove this and any other comments from the codebase. Try to write clean and readable code so you don't have to add comments to your code.

Leon-Mbegera replied:
Removed all inline comments from my codebase, so that it's clean and readable.

afanimartin commented on src/index.js:
```diff
@@ -0,0 +1,255 @@
+  return projectDiv;
+};
+
+const showProjects = () => {
```
This project comes after lessons on `Single Responsibility` and `Modules`. This module is responsible for DOM manipulation logic and JS business logic.
You need to separate DOM manipulation and JS logic. So move this function to the `project.js` module since it has logic directly related to the `project.js` module.

Leon-Mbegera replied:
Separated logic between the modules.

afanimartin commented on src/index.js:
```diff
@@ -0,0 +1,255 @@
+  });
+};
+
+// function to create a project card
```
Remove all comments

Leon-Mbegera replied:
All comments are now removed from the codebase.

afanimartin commented on src/index.js:
```diff
@@ -0,0 +1,255 @@
+
+// function to create a project card
+
+const projectExists = (projectName) => {
```
Move this to the `project.js` module.

Leon-Mbegera replied:
Moved projectExists to project.js

afanimartin commented on src/index.js:
```diff
@@ -0,0 +1,255 @@
+  return allProjects.find((project) => project.title === projectName);
+};
+
+showProjects();
```
Move this to the `project.js` module.

Leon-Mbegera replied:
Moved showProjects to project.js

afanimartin commented on src/index.js:
```diff
@@ -0,0 +1,255 @@
+
+showProjects();
+
+const addProjectToLocalStorage = (newProject) => {
```
Move this to the `project.js` module.

afanimartin commented on src/project.js:
```diff
@@ -0,0 +1,20 @@
+    this.todos.push(todo);
+  }
+
+  getTodo = (todoTitle) => {
```
These functions are directly related to the `todo.js` module so move them there.

Leon-Mbegera replied:
Moved addTodo to the respective module.

afanimartin commented on src/index.js:
```diff
@@ -0,0 +1,255 @@
+  return modifyTodoForm;
+};
+
+const showTodos = (currentPrj, newTodoCreationForm) => {
```
Move this function to the `todo` module.

Leon-Mbegera replied:
Moved showTodos to the todo module.

afanimartin commented on src/index.js:
```diff
@@ -0,0 +1,255 @@
+  return allProjects;
+};
+
+const modifyTodo = (todo, showTodos, newTodoCreationForm) => {
```
This function should live in the `todo` module so move it there.

Leon-Mbegera replied:
Moved modifyTodo to todo module.

afanimartin commented on src/index.js:
```diff
@@ -0,0 +1,255 @@
+const projectTodos = document.getElementById('todo-creation-form');
+
+// Loop thru each project and display it
+const getProjectsFromLocalStorage = () => {
```
Move this to the `project` module.

Leon-Mbegera replied:
Moved getProjectsFromLocalStorage function to dependency module.

afanimartin commented on package.json:
```diff
@@ -0,0 +1,31 @@
+    "eslint-config-airbnb-base": "^14.2.1",
+    "eslint-plugin-import": "^2.24.2",
+    "webpack": "^5.50.0",
+    "webpack-cli": "^4.8.0"
```
Add `webpack-dev-server` so one can run `npm run start` without the error: `sh: 1: webpack-dev-server: not found`

Leon-Mbegera replied:
Done.

afanimartin reviewed (changes requested):
## changes required 🙏🏿 🙏🏿 🙏🏿 
Nice work on this project. You have done well in setting up the project.

### highlights 🌵 🌵 🌵 
- PR is well described
- Readme looks good

### changes 🔕 🔕 🔕 
- separate DOM logic from JS logic
- add necessary packages to the `package.json`. check comments.
- remove comments from the codebase. check comments.
- see comments for more suggestions
- I couldn't run the app because `webpack-dev-server` was not added to the `package.json` file. please fix the error.

### Martin Afani
#### Slack: whiz25

Leon-Mbegera committed:
Clean up code base by removing inline comments

Leon-Mbegera committed:
Remove all unnecessary comments from the codebase

Leon-Mbegera committed:
Move all related functions to their respective modules

Leon-Mbegera committed:
Resolve all project and todo creation forms

Leon-Mbegera committed:
Put constraint to ensure a todo is created with all required details

Leon-Mbegera committed:
Commit previous changes before merging work from changes-branch

Leon-Mbegera committed:
Commit previous changes before merging work from changes-branch

Leon-Mbegera committed:
Merge branch 'address-changes' into todos

Leon-Mbegera committed:
Make sure a user doesn't leave out any field when updating a task

Leon-Mbegera committed:
Create a new project and store it in local storage

Leon-Mbegera committed:
Post every todo edit record to local storage

Leon-Mbegera committed:
Resolve to a working applicatio after resolving dependencies

Leon-Mbegera committed:
Define my modifyTodo method in every place i'm using it

Leon-Mbegera committed:
Resolve webpack issue to run npm start

Leon-Mbegera commented:
In these requested changes, I have,
- Moved all public methods to their respective modules
- Added a dependency file to hold common functions and variables
- Modified local storage to record every addition, update or delete done on project todos.

mahmoud717 reviewed (approved):
### STATUS: APPROVED ✔️ ✔️ 

**hi @Leon-Mbegera**

**Great job with making all of the requirements. the project is approved, you can go ahead and merge this pull request**
